sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java

```diff
@@ -203,21 +422,160 @@ public FormRecognizerServiceVersion getServiceVersion() {
         fetchExtractReceiptResult(boolean includeTextDetails) {
         return (pollingContext) -> {
             final UUID resultUid = UUID.fromString(pollingContext.getLatestResponse().getValue().getResultId());
             return service.getAnalyzeReceiptResultWithResponseAsync(resultUid)
-                .map(modelSimpleResponse -> toReceipt(modelSimpleResponse.getValue().getAnalyzeResult(),
-                    includeTextDetails));
+                .map(modelSimpleResponse -> {
+                    throwIfAnalyzeStatusInvalid(modelSimpleResponse);
+                    return toReceipt(modelSimpleResponse.getValue().getAnalyzeResult(),
+                        includeTextDetails);
+                });
+        };
+    }
+
+    private Function<PollingContext<OperationResult>, Mono<OperationResult>> contentAnalyzeActivationOperation(
+        String sourceUrl) {
+        return (pollingContext) -> {
+            try {
+                return service.analyzeLayoutAsyncWithResponseAsync(new SourcePath().setSource(sourceUrl))
+                    .map(response ->
+                        new OperationResult(parseModelId(response.getDeserializedHeaders().getOperationLocation())));
+            } catch (RuntimeException ex) {
+                return monoError(logger, ex);
+            }
+        };
+    }
+
+    private Function<PollingContext<OperationResult>, Mono<OperationResult>> contentStreamActivationOperation(
+        Flux<ByteBuffer> buffer, long length, FormContentType formContentType) {
+        return (pollingContext) -> {
+            try {
+                return service.analyzeLayoutAsyncWithResponseAsync(ContentType.fromString(formContentType.toString()),
+                    buffer, length)
+                    .map(response -> new OperationResult(
+                        parseModelId(response.getDeserializedHeaders().getOperationLocation())));
+            } catch (RuntimeException ex) {
+                return monoError(logger, ex);
+            }
+        };
+    }
+
+    private Function<PollingContext<OperationResult>, Mono<PollResponse<OperationResult>>>
+        extractContentPollOperation() {
+        return (pollingContext) -> {
+            PollResponse<OperationResult> operationResultPollResponse = pollingContext.getLatestResponse();
+            String modelId = operationResultPollResponse.getValue().getResultId();
+            try {
+                UUID resultUid = UUID.fromString(modelId);
+                return service.getAnalyzeLayoutResultWithResponseAsync(resultUid)
+                    .flatMap(modelSimpleResponse -> processAnalyzeModelResponse(modelSimpleResponse,
+                        operationResultPollResponse));
+            } catch (HttpResponseException e) {
+                logger.logExceptionAsError(e);
+                return Mono.just(new PollResponse<>(LongRunningOperationStatus.FAILED, null));
+            }
```

Comment:
Any reason this handles the exception case differently?

Reply:
This is consistent all across. The other way of handling exceptions is when the service returns a 200 OK and creates a response with the model returning an invalid status.